khronicle-core/src/commonMain/kotlin/Metadata.kt

```diff
@@ -1,21 +1,31 @@
 package com.juul.khronicle
 
-internal class Metadata : ReadMetadata, WriteMetadata {
+import kotlin.reflect.KClass
+
+/**
+ * Use this class for writing tests only! Otherwise, library consumers should operate on the
+ * [ReadMetadata] and [WriteMetadata] interfaces.
```

Comment:
Do/will we be using this class for purposes other than testing?
If it truly should only be used for tests, and never for general use, then we should probably introduce a `@KronicleInternal` annotation? That way, in tests, consumers can opt-in, but then we eliminate accidental usages.

Reply:
`@KhronicleInternal` (this annotation actually already exists) didn't feel appropriate because I intend to use this in a downstream/closed-source test class.

Reply:
I think it ultimately comes down to: if you want to maintain the API as-is. If you think (or expect) it might change, then `@KhronicleInternal` seems appropriate (and we can opt-in downstream). If you're happy with it being a part of the public API, then I'm fine w/o the annotation.

Reply:
The KDoc does seem kind of strange though. Seems like you're saying it shouldn't be a part of the public API, but not using an annotation to express that. 🤷

Reply:
Well, I think we lack an appropriate annotation. What this really needs is something more akin to `@VisibleForTesting`, but AFAIK that's only available in JVM (through JetBrains/java-annotations) or Android (as a built in). The behavior I'm looking for is public API since I want consumers to be able to access it -- I just don't want it used outside of `@Test` code.

Reply:
I'll try and make the comment better 😅

Reply:
I think I have an idea for how to fix this. I'll re-request review in a bit.